src/routes/ordersRouter.ts

```diff
@@ -0,0 +1,75 @@
+import express, { NextFunction, Request, Response } from "express";
+import { IFullOrder, IOrder, IProduct } from "../db/modelsType";
+import { orderService, productService } from "../services";
+
+const ordersRouter = express.Router();
+
+ordersRouter.get(
+  "/:id",
+  async function (req: Request, res: Response, next: NextFunction) {
+    const userId = Number(req.params.id);
+    try {
+      const ordersList: IOrder[] = await orderService.getOrdersByUserId(userId);
+      const products: IProduct[] = await productService.getProducts();
+      let userOrders: IFullOrder[] = [];
+      ordersList.map((ord) => {
+        for (let i = 0; i < products.length; i++) {
+          if (products[i].id === ord.ProductId) {
+            userOrders = [
+              ...userOrders,
+              {
+                orderId: ord.id,
+                productId: products[i].id,
+                productName: products[i].name,
+                productImg: products[i].img,
+                price: products[i].price,
+                currency: products[i].currency,
+              },
+            ];
+          }
```

Comment:
Here products loop will always walk through all products. Better to `find` product: `const product = products.find(({ id }) => id === order.productId)`.